### PR TITLE
feat(octavia): load balancer creation without listeners

### PR DESCRIPTION
**Pull request by qpavy**
ref: MANAGER-12079

| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/octavia-batch-1`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-12079
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- ~~[ ] Breaking change is mentioned in relevant commits~~

### CODE REVIEW

File: packages/manager/modules/octavia-load-balancer/src/create/service.js [hunk 1]
```diff
@@ -39,4 +42,73 @@ export default class OctaviaLoadBalancerCreateService {
...
+    return this.$http
+      .get(
+        `/cloud/project/${projectId}/region/${regionName}/loadbalancing/flavor`,
+      )
+      .then(
+        ({ data }) =>
+          data.find(
+            (regionalizedFlavors) =>
+              regionalizedFlavors.name === size.technicalName,
+          ).id,
```
Comment (darsene) [inline packages/manager/modules/octavia-load-balancer/src/create/service.js:111]:
```suggestion
          )?.id,
```
To avoid JS error

**Review by JacquesLarique — APPROVED** [review]

File: packages/manager/modules/octavia-load-balancer/src/translations/Messages_fr_FR.json [hunk 2]
```diff
@@ -1,3 +1,4 @@
 {
-  "octavia_load_balancer": "Load Balancer"
+  "octavia_load_balancer": "Load Balancer",
+  "octavia_load_balancer_global_error": "L'erreur suivante est arrivée : <br> {{message}}<br> request_id: {{requestId}}"
```
Comment (darsene) [inline packages/manager/modules/octavia-load-balancer/src/translations/Messages_fr_FR.json:3]:
```suggestion
  "octavia_load_balancer_global_error": "L'erreur suivante est survenue : <br> {{message}}<br> request_id: {{requestId}}"
```
In french, `une erreur est arrivée` is a bit weird

File: packages/manager/modules/octavia-load-balancer/src/components/instances-table/instances-table.html [hunk 3]
```diff
@@ -7,137 +7,139 @@
...
+                    <span
+                        class="oui-icon oui-icon-help"
+                        oui-tooltip="{{:: 'octavia_load_balancer_instances_table_pools_tooltip' | translate }}"
+                    ></span>
+                </th>
+            </tr>
+        </thead>
+        <tbody>
+            <tr
+                data-ng-repeat="listener in $ctrl.listeners track by $index"
```
Comment (darsene) [inline packages/manager/modules/octavia-load-balancer/src/components/instances-table/instances-table.html:33]: No side effect if the array is not always sort in the same way ?

File: packages/manager/modules/octavia-load-balancer/src/create/controller.js [hunk 4]
```diff
@@ -47,6 +61,8 @@ export default class OctaviaLoadBalancerCreateCtrl {
       loadBalancerInstance: { name: 'load_balancer_instance', display: null },
       loadBalancerName: { name: 'load_balancer_name', display: null },
     };
+
+    this.currentStep = 0;
```
Comment (darsene) [inline packages/manager/modules/octavia-load-balancer/src/create/controller.js:65]: **niptick** : put it in a function as it's used in different places. 
```suggestion
    this.resetCurrentStep();
```

File: packages/manager/modules/octavia-load-balancer/src/create/controller.js [hunk 5]
```diff
@@ -115,8 +131,50 @@ export default class OctaviaLoadBalancerCreateCtrl {
       });
   }
 
+  onCancel() {
+    this.currentStep = 0;
```
Comment (darsene) [inline packages/manager/modules/octavia-load-balancer/src/create/controller.js:135]:
```suggestion
    this.resetCurrentStep();
```

**Review by darsene — CHANGES_REQUESTED** [review]

File: packages/manager/modules/octavia-load-balancer/src/components/instances-table/instances-table.html (hunk 3 above)
Comment (qpavy) [reply packages/manager/modules/octavia-load-balancer/src/components/instances-table/instances-table.html:33]: nop the different lines have no dependencies with each others

File: packages/manager/modules/octavia-load-balancer/src/create/service.js [hunk 6]
```diff
@@ -39,4 +42,73 @@ export default class OctaviaLoadBalancerCreateService {
...
+    gateway,
+    // listeners,
+    name,
+  ) {
+    const networkInformation = {
+      networkId: privateNetwork.id,
+      subnetId: subnet.id,
+    };
+
+    if (!gateway?.length) {
```
Comment (anooparveti) [inline packages/manager/modules/octavia-load-balancer/src/create/service.js:61]: **question:** Checking for `gateway.length` but assigning an object. Should it be an object or an array?

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/modules/octavia-load-balancer/src/create/controller.js [hunk 7]
```diff
@@ -115,8 +134,49 @@ export default class OctaviaLoadBalancerCreateCtrl {
...
+      this.model.loadBalancerName,
+    )
+      .then(() => {
+        this.atInternet.trackPage({
+          name: TRACKING_LOAD_BALANCER_CREATION_SUBMIT_SUCCESS,
+        });
+
+        this.$state.go('octavia-load-balancer', {
+          displayCreationBanner: true,
+        });
```
Comment (anooparveti) [inline packages/manager/modules/octavia-load-balancer/src/create/controller.js:173]: **nitpick:** Avoid injecting `$state` into controller. Instead provide a callback from `$state.resolve` and use the callback function.

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/modules/octavia-load-balancer/src/components/instances-table/instances-table.html [hunk 8]
```diff
@@ -7,137 +7,139 @@
...
+                </th>
+                <th class="oui-table__header"></th>
+                <th class="oui-table__header">
+                    <span
+                        data-translate="octavia_load_balancer_instances_table_pools"
+                    ></span>
+                    <span
+                        class="oui-icon oui-icon-help"
+                        oui-tooltip="{{:: 'octavia_load_balancer_instances_table_pools_tooltip' | translate }}"
+                    ></span>
```
Comment (anooparveti) [inline packages/manager/modules/octavia-load-balancer/src/components/instances-table/instances-table.html:27]:
```suggestion
                    <span
                        class="oui-icon oui-icon-help"
                        aria-hidden="true"
                        oui-tooltip="{{:: 'octavia_load_balancer_instances_table_pools_tooltip' | translate }}"
                    ></span>
```

File: packages/manager/modules/octavia-load-balancer/src/create/service.js (hunk 6 above)
Comment (qpavy) [reply packages/manager/modules/octavia-load-balancer/src/create/service.js:61]: If there is no gateways, I should create 1 as a object. So this is correct

File: packages/manager/modules/octavia-load-balancer/src/components/instances-table/instances-table.html [hunk 9]
```diff
@@ -7,137 +7,139 @@
...
+                                    class="oui-label d-flex align-items-center mb-1"
+                                >
+                                    <small
+                                        data-translate="octavia_load_balancer_instances_table_protocol_label"
+                                        class="mr-1"
+                                    ></small>
+                                    <span
+                                        class="oui-icon oui-icon-help"
+                                        oui-tooltip="{{:: 'octavia_load_balancer_instances_table_protocol_tooltip' | translate }}"
+                                    ></span>
```
Comment (anooparveti) [inline packages/manager/modules/octavia-load-balancer/src/components/instances-table/instances-table.html:51]:
```suggestion
                                    <span
                                        class="oui-icon oui-icon-help"
                                        aria-hidden="true"
                                        oui-tooltip="{{:: 'octavia_load_balancer_instances_table_protocol_tooltip' | translate }}"
                                    ></span>
```

File: packages/manager/modules/octavia-load-balancer/src/components/instances-table/instances-table.html [hunk 10]
```diff
@@ -7,137 +7,139 @@
...
+                                </label>
+                                <oui-select
+                                    data-name="select-listener-protocol"
+                                    data-model="$ctrl.listeners[$index].protocol"
```
Comment (anooparveti) [inline packages/manager/modules/octavia-load-balancer/src/components/instances-table/instances-table.html:55]: **nitpick:** You can directly use listener no?

File: packages/manager/modules/octavia-load-balancer/src/components/instances-table/instances-table.html [hunk 11]
```diff
@@ -7,137 +7,139 @@
...
+                                    class="oui-label d-flex align-items-center mb-1"
+                                >
+                                    <small
+                                        data-translate="octavia_load_balancer_instances_table_port_label"
+                                        class="mr-1"
+                                    ></small>
+                                    <span
+                                        class="oui-icon oui-icon-help"
+                                        oui-tooltip="{{:: 'octavia_load_balancer_instances_table_port_tooltip' | translate }}"
+                                    ></span>
```
Comment (anooparveti) [inline packages/manager/modules/octavia-load-balancer/src/components/instances-table/instances-table.html:74]:
```suggestion
                                    <span
                                        class="oui-icon oui-icon-help"
                                        aria-hidden="true"
                                        oui-tooltip="{{:: 'octavia_load_balancer_instances_table_port_tooltip' | translate }}"
                                    ></span>
```

File: packages/manager/modules/octavia-load-balancer/src/components/instances-table/instances-table.html [hunk 12]
```diff
@@ -7,137 +7,139 @@
...
+                                </label>
+                                <input
+                                    class="oui-input"
+                                    type="text"
+                                    name="select-listener-port"
+                                    data-ng-model="$ctrl.listeners[$index].port"
```
Comment (anooparveti) [inline packages/manager/modules/octavia-load-balancer/src/components/instances-table/instances-table.html:80]: **nitpick:** You can directly use listener no?

**Review by anooparveti — APPROVED** [review]

**Review by darsene — APPROVED** [review]

**Pull request merged**